### ISSUE TITLE
Integrate Donate Dust plugin

### ISSUE THREAD

**Issue by imaginaryusername** [Electron-Cash/Electron-Cash#1382]
This issue is to summarize discussion about integrating https://github.com/cculianu/DonateSpareChange , a valuable optional improvement to the privacy aspects of Electron-Cash:

0) Naming: DonateSpareChange -> Donate Dust (Jonald)

1) It seem like people are generally okay with it as an opt-in box on Tools -> Optional Features, where Label Sync and Cosigner Pool are. 

2) High-limit: An upper limit that results in heavy warning seems desirable, but seems addressed already [here](https://github.com/cculianu/DonateSpareChange/blob/78e23d2650827b70d7399ea63b6bde780e2ef2c8/DonateSpareChange/qt.py#L493)

3) Defaults donation target: To make the plugin immediately usable upon activation, it seems people are enthusiastic with donations going to miner in a 1-in-1-out minimal op_return transaction if the "charities" list is not populated. With this setup the lowest amount can also get much lower than the 770 specified [here](https://github.com/cculianu/DonateSpareChange/blob/78e23d2650827b70d7399ea63b6bde780e2ef2c8/DonateSpareChange/qt.py#L736); how much lower it can go than 192 depends on the minimum size with op_return, which @markblundeberg can chime in. Even with the charity list populated , amounts lower than 740-770 (?) should still go to miner, else they can't make it out at all unless you're willing to combine them. 

The UI should display a notification that with no charities enabled, dust will be "donated" to miner. Maybe a greyed out, cannot-be-removed, always-there item "Miner" in the list? Alternatively, a simple line of text might suffice below the "destinations" box.

4) Populating the list: Now that the list is usable out of the gate, we can be more relaxed about pulling in suggestions (which may justly include Electron-cash development fund). There are several ways this can happen: 

4.i) No suggestion: List is left blank, all additions are strictly manual. Least controversy, may discourage people from donating to causes who can benefit.

4.ii) Pre-populated list, disabled: Suggest a list out of the gate, but with everything disabled.

4.iii) Pull one at a time from an internal json: Add a "suggest" button, where if clicked, pick one item randomly from an internal hidden json that is shipped with Electron-cash and add it to the list. 

4.iv) Pull from an external repo: Add a "suggest" button, where if clicked, retrieve a separately maintained list from the internet. Might require significant additional work. 

I personally don't think any of the above matters more than the plugin getting merged (and hence expanding userbase, improving privacy for everyone), but will personally favor ii) or iii) due to their higher potential to benefit causes rather than miners.

5) People with pre-existing lists in the DonateSpareChange plugin should see their lists carried over and not disrupted =)

**Comment by cculianu** [comment]
Wow.. so thorough. Thanks man!

**Comment by cculianu** [comment]
>  benefit causes rather than miners.

Yes.. that's one of the themes of the plugin and its purpose initially.  Even the icon is a hand giving love...

**Comment by fyookball** [comment]
Can the warning be overridden? I like having idiot proof solutions like the 50sat/byte limitation.

**Comment by minisat0shi** [comment]
Neat idea!
